include/pmacc/memory/buffers/HostBuffer.hpp

```diff
@@ -28,7 +28,37 @@
 
 namespace pmacc
 {
+    template <class TYPE, unsigned DIM>
+    class HostBuffer;
+
+namespace detail
+{
+    template< class TYPE >
+    container::HostBuffer< TYPE, 1u >
+    make_CartBuffer( HostBuffer<TYPE, 1u> & hb )
```

Comment:
Exactly, I wanted to do that as well.
But the methods I need to call in `HostBuffer` and even it's base class interfaces themselves lack all const-ness so this would not compile.

Reply:
Also, we return pointers here to potentially changed data in the new container. Tried to refactor it in the last 30 min and we would should then implement both `TYPE * const get...() const` and `TYPE * get()` members.

Reply:
```TYPE * const``` does not make a lot of sense (not the pointee, but the pointer is const, however it's temporary anyway so effectively same as ```TYPE *```), did you mean ```TYPE const * ```?

Reply:
> Also, we return pointers here to potentially/intentionally to-be-changed data in the new container. Tried to refactor it in the last 30 min and we would should then implement both TYPE * const get...() const and TYPE * get...() members.

It do not matter what you will do with the return type later on. `const` in C++ only defines what you can currently do with the object. Therefore the correct signature is `HostBuffer<TYPE, 1u> const & hb` because the process to create a new handle is not changing any data of the `pb`.  
Never the less as @ax3l wrote it is not easy to constify the code therefore I will merge it as it is.